python_benchmarks/objdetect_benchmark/objdetect_benchmark.py

```diff
@@ -635,20 +647,132 @@ def __check_chessboard(self, synthetic_chessboard, corners):
+        if show_detected:
+            print(corners)
+            cv.drawChessboardCorners(gray, synthetic_chessboard.pattern_size, corners, ret)
+            cv.imshow("CircleGridDetect", gray)
+            cv.waitKey(0)
         return {self.get_obj_names()[0]: ch_dist}
 
 
+class SyntheticCircleGrid(SyntheticObject):
+    def __init__(self, *, board_size, cell_img_size, circle_radius, grid_type):
+        self.board_size = board_size
+        self.circle_radius = circle_radius
+        self.grid_type = grid_type
+        if grid_type == "circle_sym_grid":
+            board_image_size = [board_size[0] * cell_img_size, board_size[1] * cell_img_size]
+            self.circle_centers = np.zeros(((board_size[0]) * (board_size[1]), 2), np.float32)
+            self.circle_centers[:, :2] = np.mgrid[0:board_size[0], 0:board_size[1]].T.reshape(-1, 2)
+            self.circle_centers *= cell_img_size
+            self.circle_centers += cell_img_size / 2
+        elif grid_type == "circle_asym_grid":
+            board_image_size = [2*board_size[0] * cell_img_size, board_size[1] * cell_img_size]
+            self.circle_centers = np.zeros(((board_size[0]) * (board_size[1]), 2), np.float32)
+            tmp = np.array(np.mgrid[0:board_size[0], 0:board_size[1]].T, dtype=np.float32)
+            tmp[1::2, :, 0] += 0.5
+            self.circle_centers[:, :2] = tmp.reshape(-1, 2)
+            self.circle_centers *= cell_img_size
+            self.circle_centers[:, 0] *= 2
+            self.circle_centers += cell_img_size / 2
+        else:
+            raise TypeError("unknown synthetic_object: " + grid_type)
+        self.image = np.full((board_image_size[1], board_image_size[0]), 255, dtype=np.uint8)
+
+        for circle in self.circle_centers:
+            cv.circle(self.image, (round(circle[0]), round(circle[1])), self.circle_radius, (0, 0, 0), -1, cv.LINE_AA)
+
+        self.fields = {"board_size": None, "circle_centers": None}
+        self.history = []
+        background = BackGroundObject(num_rows=int(self.image.shape[0] + cell_img_size),
+                                      num_cols=int(self.image.shape[1] + cell_img_size), color=255)
+        pasting_object = PastingTransform(background_object=background)
+        self.transform_object(pasting_object)
+
+    def transform_object(self, transform_object):
+        self.image = transform_object.transform_image(self.image)
+        self.circle_centers = np.array(transform_object.transform_points(self.circle_centers), dtype=np.float32)
+        if transform_object.name != "":
+            self.history.append(transform_object.name)
+        return self
+
+    def show(self, wait_key=0):
+        assert self.image is not None
+        image = np.copy(self.image)
+        circle_centers = self.circle_centers.reshape(-1, 1, 2)
+        cv.aruco.drawDetectedCornersCharuco(image, circle_centers)
+        cv.imshow("SyntheticCircleGrid", image)
```

Comment:
drawDetectedCornersCharuco is not relevant here.

Reply:
fixed